lib/Sema/CSSolver.cpp

```diff
@@ -142,7 +141,7 @@ Solution ConstraintSystem::finalize(
     if (getFixedType(tv))
       continue;
 
-    switch (allowFreeTypeVariables) {
+    switch (solverState->AllowFreeTypeVariables) {
```

Comment:
Are there still paths where we can end up in `finalize()` with `solverState` null?

If not we can remove some of the conditional checks like the one 13 lives above.

Anyway, LGTM!

Reply:
Yes, there shouldn't be any places where `finalize()` is called without `SolverState` being present, I'm going to turn that into assertion.